src/pages/studyView/StudyViewPage.tsx

```diff
@@ -126,6 +132,17 @@ export default class StudyViewPage extends React.Component<IStudyViewPageProps,
         }
     });
 
+    @computed
+    get addChartButtonText() {
+        if (!this.props.routing.location.query.tab || this.props.routing.location.query.tab === StudyViewPageTabKeys.SUMMARY) {
```

Comment:
since i see we are making a lot of reference to this, lets make getter on the store, currentTab, which then references this.  That way we won't end up with tons of referenences to router everywhere.   This will be actually be close to the ultimate solution.

Reply:
Sounds good, will do.